Use `systemd-modules-load` in place of custom driver-loading unit (#11)
ADSD-SoC-FPGA/Code: examples/passthrough/linux/module-loading/README.md

# Loading Kernel Modules

## For Testing

When testing functionality (e.g. the modules themselves, or corresponding hardware), the `insmod` command can be used to load the module:
```sh
insmod path/to/some_module.ko
```

This simply loads the specified module.

## For Regular Use

> In order to load modules on-demand, we can take advantage of automatic loading functionality that's built into Linux (or more precisely, Systemd).

Reviewer: While technically we could use the systemd service to load "on-demand" anytime, the intended and normal use is loading at boot time.
```suggestion
In order to load modules on boot, we can take advantage of [automatic module loading functionality that's built into systemd](https://www.freedesktop.org/software/systemd/man/latest/modules-load.d.html).
```

Also, if we're going to be pedantic, we should include links to the documentation, which I've added...

The steps to accomplish this are:

- Compile your kernel module as usual
  - Run the `make` command from the module's directory (for example, the [`ad1939`](../ad1939) directory)
- Install the module to your network-shared root filesystem

>   - From the same directory, run this command:

Reviewer: I think this is more clear. We have to be explicit about paths for people who are learning.
```suggestion
  - From the module's directory, run this command:
```

    ```sh

>     sudo KDIR=<path_to_linux-socfpga_repo> make INSTALL_MOD_PATH=/srv/nfs/de10nano/ubuntu-rootfs modules_install

Reviewer: Defining the `KDIR` variable here is not necessary because it is currently defined in the makefile.

Author: Good point, thanks! I did that on my own system, since my `KDIR` isn't where the Makefile defaults to, and blindly copied the command from my shell history. Will fix shortly.
